src/index.d.ts

```diff
@@ -545,165 +1005,885 @@ declare namespace admin.messaging {
+   *
+   * See
+   * [Manage topics from the server](/docs/cloud-messaging/manage-topics)
+   * for code samples and detailed documentation.
+   */
   interface MessagingTopicManagementResponse {
+
+    /**
+     * The number of registration tokens that could not be subscribed to the topic
+     * and resulted in an error.
+     */
     failureCount: number;
+
+    /**
+     * The number of registration tokens that were successfully subscribed to the
+     * topic.
+     */
     successCount: number;
+
+    /**
+     * An array of errors corresponding to the provided registration token(s). The
+     * length of this array will be equal to [`failureCount`](#failureCount).
+     */
     errors: admin.FirebaseArrayIndexError[];
   }
 
+  /**
+   * Interface representing the server response from the
+   * {@link https://firebase.google.com/docs/reference/admin/node/admin.messaging.Messaging#sendAll `sendAll()`} and
+   * {@link https://firebase.google.com/docs/reference/admin/node/admin.messaging.Messaging#sendMulticast `sendMulticast()`} methods.
+   */
   interface BatchResponse {
+
+    /**
+     * An array of responses, each corresponding to a message.
+     */
     responses: admin.messaging.SendResponse[];
+
+    /**
+     * The number of messages that were successfully handed off for sending.
+     */
     successCount: number;
+
+    /**
+     * The number of messages that resulted in errors when sending.
+     */
     failureCount: number;
   }
-
+  /**
+   * Interface representing the status of an individual message that was sent as
+   * part of a batch request.
+   */
   interface SendResponse {
+
+    /**
+     * A boolean indicating if the message was successfully handed off to FCM or
+     * not. When true, the `messageId` attribute is guaranteed to be set. When
+     * false, the `error` attribute is guaranteed to be set.
+     */
     success: boolean;
+
+    /**
+     * A unique message ID string, if the message was handed off to FCM for
+     * delivery.
+     *
+     */
     messageId?: string;
+
+    /**
+     * An error, if the message was not handed off to FCM successfully.
+     */
     error?: admin.FirebaseError;
   }
 
+  /**
+   * Gets the {@link admin.messaging.Messaging `Messaging`} service for the
+   * current app.
+   *
+   * @example
+   * ```javascript
+   * var messaging = app.messaging();
+   * // The above is shorthand for:
+   * // var messaging = admin.messaging(app);
+   * ```
+   *
+   * @return {!admin.messaging.Messaging} The `Messaging` service for the current
+   *   app.
+   */
   interface Messaging {
+    /**
+     * The {@link admin.app.App app} associated with the current `Messaging` service
+     * instance.
+     *
+     * @example
+     * ```javascript
+     * var app = messaging.app;
+     * ```
+     *
+     * @type {!admin.app.App}
+     */
     app: admin.app.App;
 
+    /**
+     * Sends the given message via FCM.
+     *
+     * @param {!admin.messaging.Message} message The message payload.
```

Comment:
Do we still need the type annotation? What happens if we remove the `{...}` annotations from all methods?